examples/pxScene2d/src/rcvrcore/AppSceneContext.js

```diff
@@ -443,97 +301,113 @@ AppSceneContext.prototype.runScriptInNewVMContext = function (packageUri, module
+        process: process,
+        require: requireMethod,
+        global: global,
+        setTimeout: function (callback, after, arg1, arg2, arg3) {
+          //pass the timers list to callback function on timeout
+          var timerId = SetTimeout(setTimeoutCallback, after, this.timers, function() { callback(arg1, arg2, arg3)});
+          this.timers.push(timerId);
+          return timerId;
+        }.bind(this),
+        clearTimeout: function (timer) {
+          var index = this.timers.indexOf(timer);
+          if (index != -1)
+          {
+            this.timers.splice(index,1);
+          }
+          ClearTimeout(timer);
+        }.bind(this),
+        setInterval: function (callback, repeat, arg1, arg2, arg3) {
+          var intervalId = SetInterval(callback, repeat, arg1, arg2, arg3);
+          this.timerIntervals.push(intervalId);
+          return intervalId;
+        }.bind(this),
+        clearInterval: function (timer) {
+          var index = this.timerIntervals.indexOf(timer);
+          if (index != -1)
+          {
+            this.timerIntervals.splice(index,1);
+          }
+          ClearInterval(timer);
+        }.bind(this),
+        importTracking: {}
+      });
+    }
+
     // - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 
     xModule.initSandbox(newSandbox);
     thisAppSceneContext.sandbox = newSandbox; //xModule.sandbox;
 
+
     try {
       //JRJRJRJR  This line causing a garbage collection leak...
       // LEAKLEAK
 //      this.innerscene.api = {isReady:false, onModuleReady:onAppModuleReady.bind(this) };
 
       var sourceCode = AppSceneContext.wrap(code);
-      //var script = new vm.Script(sourceCode, fname);
-      //var moduleFunc = script.runInNewContext(newSandbox, {filename:fname, displayErrors:true});
-      // fix debug under windows issue
-      var moduleFunc = vm.runInNewContext(sourceCode, newSandbox, {filename:path.normalize(fname), displayErrors:true});
+      log.message(4, "createModule_pxScope.call()");
+      var px = createModule_pxScope.call(this, xModule);
+      log.message(4, "createModule_pxScope.call() done");
+      if (isDuk) {
+        vm.runInNewContext(sourceCode, newSandbox, {
+          filename: path.normalize(fname),
+          displayErrors: true
+        }, px, xModule, fname, this.basePackageUri);
+      } else {
+        vm.runInNewContext(sourceCode, newSandbox, {
+          filename: path.normalize(fname),
+          displayErrors: true
+        })(px, xModule, fname, this.basePackageUri);
+      }
+      log.message(4, "vm.runInNewContext done");
 
-      if (process._debugWaitConnect) {
+      if (!isDuk && process._debugWaitConnect) {
         // Set breakpoint on module start
         if (process.env.BREAK_ON_SCRIPTSTART != 1)
           delete process._debugWaitConnect;
         const Debug = vm.runInDebugContext('Debug');
         Debug.setBreakPoint(moduleFunc, 0, 0);
```

Comment:
It looks like moduleFunc is no longer set.

Reply:
moduleFunc is not needed now - I invoke it in place, see "(px, xModule, fname, this.basePackageUri)"

Reply:
It looks like moduleFunc is still being used in line 409 (in the new code).

Reply:
ok, restored moduleFunc

Reply:
The reference to moduleFunc on line 410 `(Debug.setBreakPoint(moduleFunc, 0, 0); ` is not defined in its scope.  Unfortunately, the variable for moduleFunc goes out of scope after the else statement above.  The line `Debug.setBreakPoint(moduleFunc, 0, 0);` will result in an undefined variable of moduleFunc.

Reply:
The use of var allows the scope for moduleFunc to be valid.